packages/compass-data-modeling/src/components/diagram-editor.tsx

```diff
@@ -333,21 +367,121 @@ const DiagramContent: React.FunctionComponent<{
     [onAddFieldToObjectField]
   );
 
-  const diagramProps = useMemo(
-    () => ({
-      isDarkMode,
-      title: diagramLabel,
-      edges,
-      nodes,
-      onAddFieldToNodeClick: onClickAddFieldToCollection,
-      onAddFieldToObjectFieldClick: onClickAddFieldToObjectField,
-      onNodeClick,
-      onPaneClick,
-      onEdgeClick,
-      onFieldClick,
-      onNodeDragStop,
-      onConnect,
-    }),
+  const onSelectionChange = useCallback(
+    ({ nodes, edges }: { nodes: NodeProps[]; edges: EdgeProps[] }) => {
+      // Select the first selected item (if any) in the diagram
+      // since unlike react-flow, we only support single item selections
+      if (nodes.length > 0) {
+        onNodeClick(null, nodes[0]);
+        return;
+      }
+      if (edges.length > 0) {
+        onEdgeClick(null, edges[0]);
+        return;
+      }
+      // If nothing is selected, clear the selection
+      onDiagramBackgroundClicked(true);
+    },
+    [onDiagramBackgroundClicked, onNodeClick, onEdgeClick]
+  );
+
+  const onKeyDown = useCallback(
```

Comment:
We use `useHotkeys` elsewhere in compass for hotkey handling, for Consistency we should continue using it